tests/from_msa/test_prg_builder.py

```diff
@@ -0,0 +1,146 @@
+from pathlib import Path
+from unittest import TestCase
+
+from Bio.Seq import Seq
+from Bio.SeqRecord import SeqRecord
+
+from make_prg.from_msa.prg_builder import PrgBuilder
+from tests.from_msa import make_alignment, MSA
+
+this_dir = Path(__file__).resolve().parent
+data_dir = this_dir.parent / "data" / "make_prg_from_msa"
+
+
+class TestConsensusString(TestCase):
+    def test_all_match(self):
+        alignment = make_alignment(["AATTA", "AATTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "AATTA")
+
+    def test_mixed_match_nonmatch(self):
+        alignment = make_alignment(["AAGTA", "CATTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "*A*TA")
+
+    def test_indel_nonmatch(self):
+        alignment = make_alignment(["AAAA", "A--A"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "A**A")
+
+    def test_IUPACAmbiguous_nonmatch(self):
+        alignment = make_alignment(["RYA", "RTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "**A")
+
+    def test_N_special_treatment(self):
+        """
+        i)A and N at pos 2 are different, but still consensus
+        ii)N and N at pos 0 are same, but not consensus"""
+        alignment = make_alignment(["NTN", "NTA"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "*TA")
+
+    def test_all_gap_nonmatch(self):
+        alignment = make_alignment(["A--A", "A--A"])
+        result = PrgBuilder.get_consensus(alignment)
+        self.assertEqual(result, "A**A")
+
+
+def msas_equal(al1: MSA, al2: MSA):
+    if len(al1) != len(al2):
+        return False
+    for i in range(len(al1)):
+        if al1[i].seq != al2[i].seq:
+            return False
+        if al1[i].id != al2[i].id:
+            return False
+    return True
+
+
+class TestSubAlignments(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.alignment = MSA(
+            [
+                SeqRecord(Seq("AAAT"), id="s1"),
+                SeqRecord(Seq("C--C"), id="s2"),
+                SeqRecord(Seq("AATT"), id="s3"),
+                SeqRecord(Seq("GNGG"), id="s4"),
+            ]
+        )
+
+    def test_get_subalignment_sequence_order_maintained2(self):
```

Comment:
The name of this test function is a bit ambiguous. What does 2 mean?

Reply:
nice spot- i am renaming for more clarity